src/Presentation/ListFacetHtmlBuilder.php

```diff
@@ -33,37 +38,63 @@ public function buildHtml( FacetConfig $config, PropertyConstraints $state ): st
 	 * @return array<string, mixed>
 	 */
 	public function buildViewModel( FacetConfig $config, PropertyConstraints $state ): array {
-		$combineWithAnd = true; // TODO: use state and config defaultCombineWith
+		$combineWithAnd = $this->shouldCombineWithAnd( $config, $state );
 
 		return [
-			'toggle' => $this->buildToggleViewModel( $config, $state, $combineWithAnd ),
-			'checkboxes' => $this->buildCheckboxesViewModel( $config, $state, $combineWithAnd ),
+			'toggle' => $this->buildToggleViewModel( $combineWithAnd, $this->hasCombineWithChoice( $config ) ),
+			'checkboxes' => $this->buildCheckboxesViewModel( $config, $state ),
 			// TODO: act on config: showNoneFilter
 			// TODO: act on config: showAnyFilter
 		];
 	}
 
+	public function shouldCombineWithAnd( FacetConfig $config, PropertyConstraints $state ): bool {
+		if ( $this->hasCombineWithChoice( $config ) ) {
+			if ( $state->getOrSelectedValues() !== [] ) {
+				return false;
+			}
+
+			if ( $state->getAndSelectedValues() !== [] ) {
+				return true;
+			}
+		}
+
+		if ( ( $config->typeSpecificConfig[self::CONFIG_KEY_DEFAULT_COMBINE_WITH] ?? null ) === 'OR' ) {
```

Comment:
Should we define `AND` and `OR` as a constant? Or is it okay for this case?

Reply:
Unclear what would be gained by using a constant in this case

Reply:
Thinking about it a bit: it is rather arbitrary to use constants for the config keys and not for the values. With the small exception of CONFIG_KEY_DEFAULT_COMBINE_WITH being used multiple times. Using constants for the AND and OR values seems marginally better

Reply:
> Unclear what would be gained by using a constant in this case

The same value was used in multiple places so I'm wondering if it is worth it to define as a constant.

Reply:
I see only one usage of AND and OR. Are you including the tests in your count? In tests there is reason not to use production constants to avoid missing typos in them.